plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php

```diff
@@ -188,6 +188,20 @@ public function add_settings_page( $pages ) {
 		 * @return array
 		 */
 		public function add_settings_page_data( $pages ) {
+			/**
+			 * Filters the settings tabs array.
+			 *
+			 * @since 2.5.0
+			 *
+			 * @param array $available_pages The available pages.
+			 */
+			$available_pages = apply_filters( 'woocommerce_settings_tabs_array', array() );
```

Comment:
I am concerned (not too sure) about potential performance issues. It appears that `apply_filters` is invoked for each page. If this proves problematic, we could move this check to `get_page_data` to ensure it is only called once.

Reply:
Good call. I moved in in [move woocommerce_settings_tabs_array to get_page_data](https://github.com/woocommerce/woocommerce/pull/55467/commits/474de6457c639ebe6a18b14a755ec66837bdbdd2). Now the hook is called once.